src/LinAlg/matrix.h

```diff
@@ -937,6 +939,8 @@ namespace utl //! General utility classes and functions.
     int n1 = i1 > 1 || i1 < -1 ? this->size()/abs(i1) : this->size()-o1;
     int n2 = i2 > 1 || i2 < -1 ? nv/abs(i2) : nv-o2;
     int n  = n1 < n2 ? n1 : n2;
+    if (n == 0)
+      return 0.0;
     return cblas_sdot(n,this->ptr()+o1,i1,v+o2,i2);
```

Comment:
same, use `this->data()` instead of `ptr()`

Reply:
ok